src/ServerConfig.cc

```diff
@@ -106,11 +107,28 @@ ServerConfig::PluginInfo::PluginInfo(const std::string &_entityName,
                        const std::string &_filename,
                        const std::string &_name,
                        const sdf::ElementPtr &_sdf)
-  : dataPtr(new ServerConfig::PluginInfoPrivate(_entityName, _entityType,
-                                  _filename, _name))
+  : dataPtr(new ServerConfig::PluginInfoPrivate())
```

Comment:
Could use the opportunity to change to

```suggestion
  : dataPtr(std::make_unique<ServerConfig::PluginInfoPrivate>())
```

Reply:
Done in [afd30fb](https://github.com/gazebosim/gz-sim/pull/1352/commits/afd30fbb79c7c81cf9d7bb056179fa6cd92a0d94)